dpppt-backend-sdk/dpppt-backend-sdk-interops/README.md

```diff
@@ -0,0 +1,153 @@
+
+The key exchange is outside the scope of this documentation, and the EFGS documentation should be consulted. For countries hosting their own gateway service, a set of tools to simplify the process of creating and exchanging certificates is available on the EFGS repository.
+
+- `id`: An identifier for this gateway configuration.
+
+- `baseUrl`: The base URL pointing to the federation gateway. All routes will be relative to the base URL.
+
+- `authClientCert`: A base64 encoded String of a P12 containing the private key and the certificate used for mutual TLS communication.
+
+- `authClientCertPassword`: The password set to protect the `authClientCert` P12.
+
+- `signClientCert`: A base64 encoded String of a P12 containing the private key and the certificate used for signing the batches.
+
+- `signClientCertPassword`: The password set to protect the `signClientCert` P12.
+
+- `signAlgorithmName`: The signature algorithm used to sign the batches. Note that the algorithm _MUST_ match the key type given by the `signClientCert` (e.g. `sha256WithRSAEncryption`). The string value is directly used in the constructor of `JcaContentSignerBuilder`, hence check the `BouncyCastle` documentation for supported signature types.
+
+- `visitedCountries`: List of strings to indicate which countries a key visited. For now this all uploaded keys are marked with all countries present in `visitedCountries`, so theoretically the keys would always be distributed to all countries. This might change in the future. Though, since it is extremely hard to anonymize the list of visited countries on a individual level, the decision to distribute _all_ the key material to _all_ countries was preferred over a complex obfuscation scheme.
+
+- `defaultTransmissionRiskLevel`: This value is not used anymore and defaults to `Integer.MAX_VALUE`. Keep in mind that Java does not interpret unsigned integers, so the decimal, numerical value is `2147483647`. On a byte level this corresponds to `0x7f_ff_ff_ff`.
+
+- `defaultReportType`: When no `ReportType` is given, use `defaultReportType` instead. The default is `ReportType.CONFIRMED_TEST`.
+
+## Federation Gateway Synchronization
+
+The Interoperability Service will synchronize with the Federation Gateway service with a period defined by the cron string in the `WSSchedulingBaseConfig`. This defaults to `0 0/10 * * *`, meaning the sync starts every 10 minutes starting at 00:00. During the synchronization two events happen: first the Interoperability Service tries to download new keys for the last `n` days. Then all keys, which are allowed to be released and haven't been released before are uploaded to the Federation Gateway. Since the potential payload of such requests can be quite large, a batching logic is applied, which is further described in the next parts.
```

Comment:
mention the name of the cron property: `interops.efgs.syncCron`

Reply:
👍